judge-server/judge/progress_publisher.py

```
@@ -0,0 +1,49 @@
+            "progressPercent": progress_percent,
+            "currentTest": current_test,
+            "totalTests": total_tests,
+            "timestamp": datetime.now().isoformat()
+        }
+
+        channel = f"progress:{submission_id}"
+
+        try:
+            self.redis_client.publish(channel, json.dumps(message))
+            logger.info(f"Published progress for {submission_id}: {status} {progress_percent}%")
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)